mpxe/harness/pm.py

```diff
@@ -0,0 +1,82 @@
+import os
+import threading
+import select
+import subprocess
+import signal
+
+from mpxe.harness import project
+from mpxe.harness import canio
+from mpxe.sims.sim import Sim
+
+class ProjectManager:
+    def __init__(self):
+        # index projects by stdout and ctop_fifo fd
+        self.proj_fds = {}
+        self.statuses = {}
+        self.proj_name_list = os.listdir('/home/vagrant/shared/firmware_xiv/projects')
```

Comment:
stored root as a constant in pm.py but it makes imports wonky, would appreciate suggestions since I'm assuming it's best to use that constant then build off it in other modules